app/test.hoon

```diff
@@ -23,69 +23,90 @@
+::
 ++  peek  _~
+::
 ++  report-error
-  |=  [a=spur b=(each cage tang)]  ^-  tang
-  =/  should-fail  (~(get by failing) (flop a))
-  ?-    -.b
-      %&
+  |=  [=spur bud=build-result]
+  ^-  tang
+  =/  should-fail  (~(get by failing) (flop spur))
+  ?-    -.bud
+      %success
     ?~  should-fail  ~
     :~  leaf+"warn: expected failure, {<`tape`u.should-fail>}"
         leaf+"warn: built succesfully"
-        (sell q.p.b)
+        ?:  ?=(%bake +<.bud)
+          (sell q.cage.bud)
+        ?>  ?=(%core +<.bud)
+        (sell vase.bud)
     ==
   ::
-      %|
+      %error
     ?^  should-fail
       ~[>[%failed-known `tape`(weld "TODO: " u.should-fail)]<]
-    (flop p.b)
+    (flop message.bud)
   ==
 ::
 ++  made-a-core
-  |=  {a/spur @uvH b/gage}
+  |=  [=spur @da res=made-result]
   :_  +>.$
-  ?>  ?=([%tabl [(each) (each)] ~] b)
-  %-  (slog (report-error a p.i.p.b))
-  =/  nex/(list spur)
-    =<(p ;;(,[%& %cont * p=(list spur)] q.i.p.b))
+  ?:  ?=([%incomplete *] res)
+    ~&  incomplete-core+spur
+    ((slog tang.res) ~)
+  ?.  ?=([%complete %success *] res)
+    ~&  unsuccessful-core+spur
+    ((slog message.build-result.res) ~)
+  ?>  ?=(^ +<.build-result.res)
+  %-  (slog (report-error spur head.build-result.res))
+  =/  nex=(list ^spur)
+    =<  p
+    ;;  ,[%success %$ %cont * p=(list ^spur)]
+    tail.build-result.res
   ?~  nex  ~&(%cores-tested ~)
   [ost (build-core nex)]~
 ::
 ++  build-core
   |=  [a=spur b=(list spur)]
   ~&  >>  (flop a)
-  :^  %exec  a-core+a  our
-  %-  some
-  ^-  bilk
-  :-  now-beak
-  :~  %tabl
-    [[%core now-beak a] [%$ %cont !>(b)]]
-  ==
+  :-  %build
+  :^    a-core+a
+      our
+    live=|
+  ^-  schematic:ford
+  :-  [%core now-disc %hoon a]
+  [%$ %cont !>(b)]
 ::
 ++  made-a-rend
-  |=  {a/wire @uvH b/gage}
-  ?>  ?=([ren=mark ~] a)
-  =+  `[ren=term pax=path]`?~(a !! a)
+  |=  [=spur @da res=made-result]
   :_  +>.$
-  ?>  ?=([%tabl [(each) (each)] ~] b)
-  %-  (slog (report-error /[ren.i.a]/ren p.i.p.b))
-  =/  nex/(list term)
-    =<(p ;;(,[%& %cont * p=(list term)] q.i.p.b))
+  ?>  ?=([ren=term ~] spur)
+  =+  `[ren=term pax=path]`?~(spur !! spur)
+  ?:  ?=([%incomplete *] res)
+    ~&  incomplete-core+spur
+    ((slog tang.res) ~)
+  ?.  ?=([%complete %success *] res)
+    ~&  unsuccessful-core+spur
+    ((slog message.build-result.res) ~)
+  ?>  ?=(^ +<.build-result.res)
+  %-  (slog (report-error /[ren]/ren head.build-result.res))
+  =/  nex=(list term)
+    =<  p
+    ;;  ,[%success %$ %cont * p=(list term)]
+    tail.build-result.res
   ?~  nex  ~&(%rens-tested ~)
   [ost (build-rend nex)]~
 ::
 ++  build-rend
   |=  [a=term b=(list term)]
   ~&  >>  [%ren a]
-  =/  bem/beam  (need (de-beam %/example))
-  =.  -.bem  now-beak
-  :^  %exec  a-rend+/[a]  our
-  %-  some
-  ^-  bilk
-  :-  -.bem
-  :~  %tabl
-    [`silk`[%bake a fake-fcgi bem] [%$ %cont !>(b)]]
-  ==
+  :-  %build
+  :^    a-rend+/[a]
+      our
+    live=|
+  ^-  schematic:ford
+  :: XX what should the rail be?
```

Comment:
```
=/  bem=beam  (need (de-beam %/example))
=/  =rail  [[p q] s]:bem
```

Reply:
That makes sense, but I get the same errors regardless (the only difference being `%/app/test/hoon/example/hoon` in place of `%/example/hoon`):

```
> :test [%renders /]
[%findining-hoons under=/ren]
> [/ren/collections %skipped "temporarily disabled"]
> [/ren/css %skipped "not meant to be called outside /web/pack"]
> [/ren/js %skipped "not meant to be called outside /web/pack"]
/ren/rss-xml
> [/ren/run %skipped "not meant to be called except on a (different) hoon file"]
/ren/test-gen
/ren/tree/index
> [/ren/urb %skipped "temporarily disabled"]
> [/ren/x-collections-json %skipped "temporarily disabled"]
> [/ren/x-collections-snip %skipped "temporarily disabled"]
> [/ren/x-htm %skipped "temporarily disabled"]
> [/ren/x-urb %skipped "temporarily disabled"]
>> [%ren %rss-xml]
>> [%ren %test-gen]
{%alts: all options failed}
as-mark
{/! failed:   ford: %hood failed for /~zod/home/0/app/test/hoon/example/hoon:  scry failed for  %cx /~zod/home/~2018.8.28..07.53.16..ba0a/app/test/hoon/example/hoon}
as-renderer
ford: %bake %test-gen on /~zod/home/0/app/test/hoon/example failed:
>> [%ren %tree-index]
{%alts: all options failed}
as-mark
{/tree-elem/ failed:   ford: %bake %tree-elem on /~zod/home/0/app/test/hoon/example failed:  %alts: all options failed}
as-renderer
ford: %bake %tree-index on /~zod/home/0/app/test/hoon/example failed:
%rens-tested
```

I wonder if the `%bake` semantics have changed in case of a scry failure, or if this is a more general change across ford. In any case, given the number of renderers that are already excluded from testing, it might be fine to just disable them all until a more robust approach can be developed for them.

Reply:
Oh, I think what happened here is that this tried to run the test renderer at ren/test-gen.hoon. That renderer tries to compile .hoon file full of tests at its path to render, but the example file is a .umd file, so when the test-gen renderer tries to render the example path, it doesn't find a hoon file there.

We don't need to test the test renderer. Quis probabit ipsos probatores?